examples/plot_knockoffs_wisconsin.py

```diff
@@ -0,0 +1,237 @@
+repeats_noise = 5  # Number of synthetic noisy sets to add
+
+noises_train = [X_train]
+noises_test = [X_test]
+feature_names_noise = [x for x in feature_names]
+for k in range(repeats_noise):
+    X_train_c = X_train.copy()
+    X_test_c = X_test.copy()
+    noises_train.append(X_train_c + 2 * rng.randn(n_train, p))
+    noises_test.append(X_test_c + 2 * rng.randn(n_test, p))
+    feature_names_noise += [f"spurious #{k*p+i}" for i in range(p)]
+
+noisy_train = np.concatenate(noises_train, axis=1)
+noisy_test = np.concatenate(noises_test, axis=1)
+
+
+#################################################################################
+# There are 180 features, 30 of them are real and 150 of them are fake and independent
+# of the outcome. We now apply the Lasso (with cross-validation to select the best
+# regularization parameter) to the noisy dataset and observe the results:
+lasso_noisy = LogisticRegressionCV(
+    Cs=np.logspace(-3, 3, 10), penalty="l1", solver="liblinear", random_state=rng
+)
+lasso_noisy.fit(noisy_train, y_train)
+y_pred_noisy = lasso_noisy.predict(noisy_test)
+print(
+    f"Accuracy of Lasso on test set with noise: {lasso_noisy.score(noisy_test, y_test):.3f}"
+)
+
+selected_mask = [
+    "selected" if np.abs(x) > 1e-6 else "rejected" for x in lasso_noisy.coef_[0]
+]
+df_lasso_noisy = pd.DataFrame(
+    {
+        "score": np.abs(lasso_noisy.coef_[0]),
+        "variable": feature_names_noise,
+        "selected": selected_mask,
+    }
+)
+# Count how many selected features are actually noise
+num_false_discoveries = np.sum(
+    np.array(selected_mask[p:]) == "selected"
+)  # Count the number of selected spurious variables
+print(f"The Lasso makes at least {num_false_discoveries} False Discoveries!!")
+
+
+#################################################################################
+# The Lasso selects many spurious variables that are not directly related to the outcome.
+# To mitigate this problem, we can use one of the statistically controlled variable
+# selection methods implemented in hidimstat. This ensures that the proportion of False
+# Discoveries is below a certain bound set by the user in all scenarios.
+
+
+#############################################################################
+# Controlled variable selection with Knockoffs
+# ----------------------------------------------------
+# We use the Model-X Knockoff procedure to control the FDR (False Discovery Rate). The
+# selection of variables is based on the Lasso Coefficient Difference (LCD) statistic
+# :footcite:t:`candes2018panning`.
+from hidimstat import model_x_knockoff
+
+fdr = 0.2
+
+selected, test_scores, threshold, X_tildes = model_x_knockoff(
+    noisy_train,
+    y_train,
+    estimator=LogisticRegressionCV(
+        solver="liblinear",
+        penalty="l1",
+        Cs=np.logspace(-3, 3, 10),
+        random_state=rng,
+    ),
+    n_bootstraps=1,
+    random_state=0,
+    preconfigure_estimator=None,
+    fdr=fdr,
+    n_jobs=3,
```

Comment:
The definition of n_jobs is unnecessary in this context.

Reply:
done

Reply:
in this case, the n_job is define by the number of bootstrap.